app/helpers/generic_file_helper.rb

```diff
@@ -39,7 +43,22 @@ def download_path(*args)
       gf = GenericFile.find(item.id)
       return gf.doi_url if gf.doi_url.present?
     end
-    return sufia.download_path(*args)
+    if args[1].nil?
```

Comment:
Careful about this.  @mbarnett already pointed out that one of the possible calls is a hash. See https://github.com/ualbertalib/HydraNorth/blob/f4aedb48e8deb77e9fdf71db6fe43d90ce0497bb/app/helpers/generic_file_helper.rb#L34